align/schema/constraint.py

```diff
@@ -816,151 +816,35 @@ class DoNotUseLib(SoftConstraint):
+        identify_array(bool): true/false
+        fix_source_drain(bool): true/false
+        keep_dummy_hierarchies(bool): true/false
+        merge_series_devices(bool): true/false
+        merge_parallel_devices(bool): true/false
 
     Example: ::
 
         {
-            "constraint": "AutoConstraint",
-            "isTrue": true,
-            "propagate": false
+            "constraint": "CompilerFlags",
+            "is_digital": true,
+            "keep_dummy_hierarchies": true,
+            "propagate": true
         }
     '''
-    isTrue: bool
-    propagate: Optional[bool]
-
-
-class IdentifyArray(SoftConstraint):
-    '''
-    Forbids/Alow any array identification
-
-    Args:
-        isTrue (bool): true/false.
-        propagate: Copy this constraint to sub-hierarchies
-
-    Example: ::
-
-        {
-            "constraint": "IdentifyArray",
-            "isTrue": true,
-            "propagate": false
-        }
-    '''
-    isTrue: bool
-    propagate: Optional[bool]
-
-
-class FixSourceDrain(SoftConstraint):
-    '''
-    Forbids auto checking of source/drain terminals of transistors.
-    If `True`, Traverses from power to ground and vice-versa to
-    ensure (drain of NMOS/ source of PMOS) is at higher potential.
-
-    Args:
-        isTrue (bool): true/false.
-        propagate: Copy this constraint to sub-hierarchies
-
-    Example: ::
-
-        {
-            "constraint": "FixSourceDrain",
-            "isTrue": true,
-            "propagate": False
-        }
-    '''
-    isTrue: bool
-    propagate: Optional[bool]
-
-
-class KeepDummyHierarchies(SoftConstraint):
-    '''
-    Removes any single instance hierarchies.
-
-    Args:
-        isTrue (bool): true/false.
-        propagate: Copy this constraint to sub-hierarchies
-
-    Example: ::
-
-        {
-            "constraint": "KeepDummyHierarchies",
-            "isTrue": true,
-            "propagate": false
-        }
-    '''
-    isTrue: bool
-    propagate: Optional[bool]
-
-
-class MergeSeriesDevices(SoftConstraint):
-    '''
-    Allow stacking of series devices
-    Only works on NMOS/PMOS/CAP/RES.
-
-    Args:
-        isTrue (bool): true/false.
-        propagate: Copy this constraint to sub-hierarchies
-
-    Example: ::
-
-        {
-            "constraint": "MergeSeriesDevices",
-            "isTrue": true,
-            "propagate": False
-        }
-    '''
-    isTrue: bool
-    propagate: Optional[bool]
-
-
-class MergeParallelDevices(SoftConstraint):
-    '''
-    Allow merging of parallel devices.
-    Only works on NMOS/PMOS/CAP/RES.
-
-    Args:
-        isTrue (bool): true/false.
-        propagate: Copy this constraint to sub-hierarchies
-
-    Example: ::
-
-        {
-            "constraint": "MergeParallelDevices",
-            "isTrue": true,
-            "propagate": false
-        }
-    '''
-    isTrue: bool
-    propagate: Optional[bool]
+    is_digital: Optional[bool] # Annotation and auto-constraint generation
```

Comment:
Setting default values here can reduce mystery and improve readability.

Reply:
done